app/models/kanji.rb

```diff
@@ -1,6 +1,12 @@
 class Kanji < ApplicationRecord
-  validates :strokes, presence: true
-  validates :character, presence: true, length: { is: 1 }
+  has_many :cards, dependent: :destroy
+  validates :character, presence: true, length: { is: 1 } #validate length
+  validates :grade, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, allow_nil: true
+  validates :strokes, presence: true, numericality: { greater_than: 0 }
+  validates :jlpt, inclusion: { in: 1..5 }, allow_nil: true
```

Comment:
I think non-jlpt will have a value of 0, so let's make it `{ in: 0..5 }`

Reply:
Ok, then I will delete `allow_nil: true`